File: python/particles/Particle.py
import pygame

class Particle():
    def __init__(self, screen, pos):
        self.pos = pos
        self.vel = pygame.Vector2()
        self.acc = pygame.Vector2()
        self.life = 0

        self.screen = screen

    def applyForce(self, force):
        self.acc += force
    
    def update(self):
        self.vel += self.acc
        self.pos += self.vel
        self.acc *= 0

        self.life += 1

    def show(self, color):
        pygame.draw.circle(self.screen, color, (self.pos.x, self.pos.y), 5)

File: cloth simulation/Particle.js
class Particle {
    constructor(x, y) {
        this.pos = createVector(x, y)
        this.vel = createVector(0, 0)
        this.acc = createVector(0, 0)

        this.locked = false
    }

    lock() {
        this.locked = true
    }

    applyForce(forceVector) {
        this.acc.add(forceVector)
    }

    update() {
        if (!this.locked) {
            this.vel.mult(0.99)
            this.vel.add(this.acc)
            this.pos.add(this.vel)
            this.acc.mult(0)
        }
    }

    display() {
        fill(255)
        ellipse(this.pos.x, this.pos.y, 10, 10)
    }
}